protocol/messenger_communities.go

```diff
@@ -1778,3 +1782,107 @@ func (m *Messenger) SyncCommunitySettings(ctx context.Context, settings *communi
 	chat.LastClockValue = clock
 	return m.saveChat(chat)
 }
+
+func (m *Messenger) ExtractDiscordDataFromImportFiles(filesToImport []string) (*discord.ExtractedData, map[string]*discord.ImportError) {
+
+	extractedData := &discord.ExtractedData{
+		Categories:             map[string]*discord.Category{},
+		ExportedData:           make([]*discord.ExportedData, 0),
+		OldestMessageTimestamp: 0,
+	}
+
+	errors := map[string]*discord.ImportError{}
+
+	for _, fileToImport := range filesToImport {
```

Comment:
Oh and one thought here:

I think we can optimize this even further and spin up goroutines for each file here.
By running this concurrently we should see an increase in speed.

Would do this in another iteration though. I've run this the way it is with many discord channel files and it only took a few seconds at max already.

Reply:
Yes, this is a great idea and this should probably be done in another PR. Concurrency can introduce a bunch of issues, so this is fine for now.